Source/SwiftLintFramework/Extensions/File+SwiftLint.swift

```diff
@@ -90,6 +90,22 @@ extension File {
         }
     }
 
+    public func syntaxKindsByLine(startLine: Int? = nil,
+                                  endLine: Int? = nil) -> [(Int, [SyntaxKind])] {
+        let contents = self.contents as NSString
+        let kindsWithLines = syntaxMap.tokens.map { token -> (Int, SyntaxKind) in
+            let tokenLine = contents.lineAndCharacterForByteOffset(token.offset)
+            return (tokenLine!.line, SyntaxKind(rawValue: token.type)!)
```

Comment:
@marcelofabri it might be better to eagerly create the `SyntaxKind`s by mapping twice:

```swift
let kindsWithLines = syntaxMap.tokens.map { token -> (Int, SyntaxToken) in
    let tokenLine = contents.lineAndCharacterForByteOffset(token.offset)
    return (tokenLine!.line, token)
}.filter { line, token in
    return line >= (startLine ?? 0) && line <= (endLine ?? Int.max)
}.map { (line, token) -> (Int, SyntaxKind) in
    return (line, SyntaxKind(rawValue: token.type)!)
}
```